core/src/test/java/org/kin/sendkin/core/UtilsUnitTest.java

```diff
@@ -0,0 +1,65 @@
+package org.kin.sendkin.core;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.kin.sendkin.core.model.RecipientContact;
+import org.kin.sendkin.core.view.Utils;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+
+public class UtilsUnitTest {
+
+    private final String address1 = "GCCLBNMWRUKPCWV4J6H42KAQQ6GSOG6YKRNONAT3QDOGRH3T3EQGSXJR";
+    private final String name1 = "James Bond";
+
+    private final String address2 = "GDK57YFZRKNKJTRLE3CZ2GNQCSRC5NPJQWIEMD43M7V6ZIHNARGBIMFI";
+    private final String name2 = "Mr Bond";
+
+    private RecipientContact contact1;
+    private RecipientContact contact2;
+
+
+    @Before
+    public void initMocks() {
+        contact1 = new RecipientContact(name1, address1);
+        contact2 = new RecipientContact(name2, address2);
+    }
+
+
+    @Test
+    public void parseTest() {
+        String contactStr = Utils.parseToString(contact1);
+        final RecipientContact parsedContact = Utils.parseFromString(contactStr);
+        assertTrue(contact1.equals(parsedContact));
+    }
+
+    @Test
+    public void parseArrayTest() {
+        ArrayList<RecipientContact> originContacts = new ArrayList<>();
```

Comment:
the test is a bit convoluted and hard to understand, I sense it should be separate to multiple ones.
follow the given,when,then pattern to come up with a  single use case  and better test naming

Reply:
i will alter this in the next sprint